DB/src/test/java/io/deephaven/db/v2/utils/TestKeyedArrayBackedMutableTable.java

```diff
@@ -278,10 +267,24 @@ private void waitForCompletion() throws InterruptedException, ExecutionException
         }
     }
 
-    private void handleDelayedRefresh(BaseArrayBackedMutableTable kabut) {
-        new Thread(() -> {
-            SleepUtil.sleep(200);
-            LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(kabut::refresh);
-        }).start();
+    private void handleDelayedRefresh(final BaseArrayBackedMutableTable table, final FunctionalInterfaces.ThrowingRunnable<IOException> action) throws Exception {
+        final Thread refreshThread;
+        final Semaphore gate = new Semaphore(0);
+        table.setOnPendingChange(gate::release);
+        try {
+            refreshThread = new Thread(() -> {
+                gate.acquireUninterruptibly();
+                LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(table::refresh);
+            });
+            refreshThread.start();
+            action.run();
+
+        } finally {
+            table.setOnPendingChange(null);
+        }
+        try {
+            refreshThread.join();
+        } catch (InterruptedException ignored) {
+        }
```

Comment:
Does it make sense to just not catch this exception and let junit handle it?

Reply:
Sure. Less code.

Reply:
I ended up wrapping and re-throwing, to add more documentation to the exception message. But it's not ignored anymore.